packages/bpk-component-input/src/BpkInput.js

```diff
@@ -78,7 +82,14 @@ const BpkInput = props => {
   }
   if (clearable) {
     classNames.push(getClassName('bpk-input--clearable'));
+    if (clearButtonMode === CLEAR_BUTTON_MODES.always) {
+      classNames.push(getClassName('bpk-input--clearable--persistent'));
+      clearButtonClassNames.push(
+        getClassName('bpk-input--clearable--persistent__clear-button'),
```

Comment:
This `classname` is non-bem. Selectors should follow:
```
.calculator {} // block 
.calculator--alternate {} // block modifier 
.calculator__definitions {} // element 
.calculator__definitions--expanded {} // element modifier
```

- `bpk-input--clearable--persistent`: This has two modifiers. I'd expect this to be a separate modifier altogether `bpk-input--persistent-clearable`
- `bpk-input--clearable--persistent__clear-button`: you cant have an element after a modifier. You need to a apply a modifier to the element in question i.e. `bpk-input__clear-button` + `bpk-input__clear-button--persistent-clearable`

Reply:
My above comment means that you'll need to modify the logic of how you apply class names too.